### DISCUSSION TITLE
2.2版本，本地yarn dev 后，Chat对话，本地访问，后台报错ConnectTimeout ，同一台电脑上，同样的网络环境，官方的Chat可以正常使用。

### DISCUSSION THREAD

**Discussion by xtwvf207** [ChatGPTNextWeb/NextChat#1347] [Q&A]
2.2版本，本地yarn dev 后，Chat对话，本地访问，后台报错ConnectTimeout ，同一台电脑上，同样的网络环境，官方的Chat可以正常使用。
不知道大家是否遇到过这类问题。以下，我将报错代码列出：

=========================
yarn run v1.22.18
$ yarn fetch && next dev
$ node ./scripts/fetch-prompts.mjs
[Fetch] fetching cn prompts...
[Fetch] fetching en prompts...
[Fetch] saved to ./public/prompts.json
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
warn  - You have enabled experimental feature (appDir) in next.config.mjs.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.

event - compiled client and server successfully in 5.6s (267 modules)
wait  - compiling...
event - compiled client and server successfully in 2.4s (267 modules)
wait  - compiling /page (client and server)...
wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 59.4s (4274 modules)
fatal: not a git repository (or any of the parent directories): .git
[Build Config] No git or not from git repo.
fatal: not a git repository (or any of the parent directories): .git
[Build Config] No git or not from git repo.
[Lang] failed to detect user lang.
[Lang] failed to detect user lang.
wait  - compiling /api/config/route (client and server)...
event - compiled successfully in 1763 ms (112 modules)
wait  - compiling /api/openai/[...path]/route (client and server)...
event - compiled successfully in 514 ms (116 modules)
[OpenAI Route] params2  { path: [ 'v1', 'chat', 'completions' ] }
[Auth] allowed hashed codes:  []
[Auth] got access code:
[Auth] hashed access code: d41d8cd98f00b204e9800998ecf842e
[User IP]  ::1
[Time]  2023/5/9 12:57:30
[Auth] use user api key
[Proxy]  v1/chat/completions?path=v1&path=chat&path=completions
[Base Url] https://api.openai.com
[common.ts URL]: https://api.openai.com/v1/chat/completions?path=v1&path=chat&path=completions
[OpenAI]   [TypeError: fetch failed] {
  cause:  [Error: connect ECONNREFUSED 122.10.85.4:443] {
  errno: -4078,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '122.10.85.4',
  port: 443
}
}

**Comment by darth-pika-hu** [comment]
你在国内吧？你对API的访问被拒绝了。

**Reply by xtwvf207** [reply]
可是，用着梯子呢

**Reply by xtwvf207** [reply]
同样的网络环境下，
https://chatgpt.nextweb.fun/ 官方的，我设置好自己的KEY之后，是可以正常使用的。
而我本地的不行。
日志中的IP【122.10.85.4】是中国香港的。

**Reply by zhuSilence** [reply]
香港的 IP 也是访问不了的，估计代理没开全局
